server/src/main/java/org/elasticsearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -168,11 +169,6 @@ boolean validate(Metadata metadata) {
         return true;
     }
 
-    @Override
-    public Iterator<IndexShardRoutingTable> iterator() {
-        return shards.values().iterator();
-    }
```

Comment:
Have you considered doing something like `Arrays.asList(shards)` to still provide iterator to minimize amount of fixes needed? Is it negating all benefits from using array instead of map?

Reply:
Yea, I bet that might give a speedup as well. But I figured I'd have to benchmark that and it's likely slower by a non-trivial margin (escape analysis could impress me here ... but not counting on it) because we mostly have just one element so the cost of setting up the iterator object will be non-trivial.
-> I'd rather go all the way here given that multiple of the spots I adjusted are very prominent in profiling.